common/plot_agent.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+// Third-party includes
+#include "../third_party/matplotlibcpp.h"
+#include "../third_party/units.h"
+
+// Standard C++ includes
+#include <array>
+
+namespace BoBRobotics {
+    template<typename AgentType>
+    void plotAgent(const AgentType &agent, std::array<double, 2> &&xlim, std::array<double, 2> &&ylim)
```

Comment:
How come you're passing these as an rvalue reference? isn't this just a case for ``const std::array<double, 2> &xlim``?

Reply:
Hmm yeah it probably should be :-/